semgrep_output_v1.atd

```diff
@@ -1617,12 +1617,15 @@ type parsing_stats = {
   num_bytes: int;
 }
 
+
 (* Response by the backend to the CLI to the POST /complete *)
 type ci_scan_complete_response <ocaml attr="deriving show"> = {
    success: bool;
    ~app_block_override: bool;
    (* only when app_block_override is true *)
    ~app_block_reason: string;
+   (* match_based_ids of findings that semgrep-app determined should cause the scan to block *)
+   ~app_blocking_match_based_ids : string list;
```

Comment:
we should define a separate string wrap type for match_based_id, just like we have a special
type for fpath, uri, rule_id, etc. It's not good to use string for everything.

Reply:
sure

Reply:
hm actually, doing that seems not super useful since there's no special internal representation for them. They're just strings and all we ever do with them is compare for equality

Reply:
It does not matter. Filenames are also strings, but we still want a different type for them. Otherwise they will be nothing that prevents a filename to go inside a match_based_id.
Types!

Reply:
ok yeah I agree